src/index.ts

```diff
@@ -2,8 +2,61 @@
  * Contains request routing logic.
  */
 
+import {HeaderKeys} from './headers';
+import * as injectorsCache from './injectors-cache';
 import router from './router';
 
+/** */
+async function cacheAfterServing(
+  event: FetchEvent,
+  response: Response
+): Promise<void> {
+  const xCacheAfterServing = response.headers.get(
+    HeaderKeys.X_CACHE_AFTER_SERVING
+  );
+  if (!xCacheAfterServing) {
+    return;
+  }
+
+  // Strip this internal header from the response being cached, or risk an
+  // infinite loop of saving to cache.
+  response.headers.delete(HeaderKeys.X_CACHE_AFTER_SERVING);
```

Comment:
We should delete the header from the initial response, too.

Reply:
Done 22441c9